### DISCUSSION TITLE
MiniDexed Shield: A(nother) MiniDexed Pcb

### DISCUSSION THREAD

**Discussion by miotislucifugis** [probonopd/MiniDexed#721] [General]
Id like to share my new version of a miniDexed PCB:

I wanted to make something that is compact and full-featured, while still keeping it as simple/ inexpensive as possible and friendly for DIY beginners.   (While I initially wanted to make my "ideal" minidexed, just for me!)  I am the repair tech at a synth museum and I often get requests to do "DIY Synth building workshops."  Usually, I turn them down because Ive never really found a circuit that is both simple enough for beginners to build in one sitting and that is also a nice, worthwhile, and playable instrument.   (Personally I find most of these simple square wave/ drone/ optical theremin/ noise maker-type synth projects rather dreadful.)   MiniDexed  has long been a contender, but I needed a solution that puts it all together in a tidy, permanent package and avoids fussy dupont jumpers and some kind of costly/impractical (at least for for class settings) 3d printed enclosure the keep everything together.   Also, it needs to keep the part count to a minimum and use (mostly*) through hole components.    The "shield" format seemed ideal  (yes, i know that for pi, they are technically called "hats", but to me they are all "shields"... anyway...)   

 Voila:

![IMG_6118](https://github.com/user-attachments/assets/a5bf3931-d535-4e4b-81f9-018341a2650f)

It uses all easy to source, off-the-shelf parts : a 1602 LCD,  a cheap and cheerful (yet impressively good ) PCM5102A DAC module, an encoder, and of course, a raspberry pi.    It features  2x 1/4" output jacks and a stereo 3.5mm jack for the Audio Out,  TRS (type A) Midi IN and OUT.    User configurable solder pads allow for a few different audio routing options: mono summing on the 1/4" jacks,  bypassing the 1/4" jacks when a 3.5mm cable is is connected,  and Direct out or Master Volume control

![IMG_6119](https://github.com/user-attachments/assets/b7016277-5209-43a1-b344-171de53db1ba)

 It can be built minimally with just the main board (+ Screen, Dac and encoder),  or with an optional "Control" expander board that  raises the encoder to make it easier for panel mounting, and/or or adding a Master Volume control.   3 different types of GPIO expander boards accomodate different types of GPIO buttons - either 5x pcb tactile switches or 4x (socketed) Kalih Choc or Cherry MX key switches.  

* The keyswitch boards have some SMD parts, but are still very doable for beginners: 4x pull up resistors, and the 4x key switch sockets)

![IMG_6131](https://github.com/user-attachments/assets/02e15be7-3351-40ab-add4-4a712dd72397)

![IMG_6121](https://github.com/user-attachments/assets/a3e9f1e6-64da-4e6e-849c-514d64f726e8)

(Im still fleshing it out, but eventually) more info ( kicad files, build guide, minidexed.ini config, etc) can be found on my github:   
https://github.com/miotislucifugis/MiniDexed-Shield

**Comment by Kirtai** [comment]
Oh this is very cool :)
Your link to your repo needs fixing though.

**Reply by miotislucifugis** [reply]
thanks!

**Comment by blixabloxa** [comment]
Looks great. Thanks for sharing.

**Comment by Gitarrenzupfer** [comment]
Very cool! Is the master volume control digital or analog. Could you please explain it in detail?

**Reply by miotislucifugis** [reply]
Its simply a dual 100k audio taper pot on the output.   Ill try to put up a jpg schematic soon.

**Reply by Gitarrenzupfer** [reply]
I found the schematic already, but I could find the value of the pot. Anyway, a jpg schematic would be helpful für anybody...
Thank you 👍

**Reply by miotislucifugis** [reply]
In general, the value is not super critical, 100k - 1m will work. Be sure to use Audio taper tho'.      Schematic, build guide etc are all on my to "do list"

**Comment by miotislucifugis** [comment]
Btw, if any one is interested in PCBs, full kits, or completed devices, just ask.  ( I may eventually put them on my reverb/etsy shops and the Discord Linear night market channel,(... another thing on my list.))   I do have a few boards from the prototype batch that have a few minor errors (nothing major)  that I would be happy to give out for free (+ shipping, of course... or you can pick them up in lovely Fribourg, Switzerland! )  You can contact me via email : charbot at gmail

**Reply by BobanSpasic** [reply]
Any possibility to add standard DIN connectors for MIDI as an add-on board?
I am interested in a PCB, possibly with pre-soldered components (excl. Pi, display and DAC)
I am located in Vienna, Austria.

**Reply by miotislucifugis** [reply]
There is no room to directly solder  DIN jacks, but you could make a daughter board  or use panel mount jacks and wire them from the 3.5 jack pads.   Send me a email and we can discuss the details.  I added my contact to the previous post.

**Reply by diyelectromusic** [reply]
That build/board looks amazing :)

I have found in my own pcbs that it is possible to overlap MIDI TRS and DIN footprints on the board as an option, but looking at your board that might make it slightly longer, and maybe a little wider...

Here's an example for my RPi 400 MiniDexed board: https://diyelectromusic.com/2023/12/18/rpi-400-midi-and-audio-pcb-design/

I can't claim credit mind - I pinched the idea from the Adafruit MIDI Feather Wing!

Kevin

**Reply by miotislucifugis** [reply]
Hey thanks! 
It would have to be longer _and_ thicker - the height of the jacks are a bit taller than the pi + header.    And then it would mess with the spacing and layout of the controls/screen which i fussed over for quite a bit.    At first, I really wanted DIN jacks and I definitely tried cramming it all in, but there was really no way w/o making it chunkier than I wanted.  Trs seemed like a fair compromise, because as a whole, it looked better, and with midi on the side + power and audio out of the back...  w/ some stout DIN cables plugged in, and 2x 1/4" cables... the whole ensemble starts getting kinda clunky.  

I have to admit tho',  Im really starting to appreciate (even love) trs midi (only type A, of course!) and have embraced it for a few other projects.     It is perfect for diminutive pocket-sized synths like this  and is great for travel / small desktop setups.   DIN-TRS adapter are pretty common nowadays and personally, I welcome the thinner( and usually shorter) cables.

**Comment by miotislucifugis** [comment]
(looks like im replying to a post that has disappeared!?)
 "I'd be very keen in building this project. Is there any news on availability of PCBs please? It appears the files in the repo are missing the PCB design and older commits appear to have an error in KiCad."


I have several pcbs still available and can provide completed units, Kits with almost all parts and bare pcbs... contact me directly if you are interested. my email is above. Ive been meaning to post them on etsy or something, but Im kinda lazy.

What are the issues with the files on github?    Ill have to double check, all the files should be there.  the only issue with the older commit was there were a few overlaps in the border edge cuts and this prevent kicad from generating a .step file.   Everything should be good otherwise for generating the gerbers.  it is what I used for the pcbs.

**Reply by maq1017** [reply]
Hi, sorry yes.. I saw later that you'd posted to contact via email, so did that instead :) I'd be very keen on either purchasing PCB or a kit if you've any available.

The latest merge on github appeared to have an empty PCB file when I last looked. I did go back to some previous commits but as you say, the border edge cuts flagged as an issue and I wasn't confident enough to proceed

**Reply by miotislucifugis** [reply]
i dont see your email,  double check the address... charbot at  gmail

**Reply by maq1017** [reply]
Seems that's what I used, assuming dot com. Strange!

**Reply by maq1017** [reply]
I've emailed again this morning to the address as above, thanks.

**Comment by matbenalex** [comment]
Hi,
I built the MiniDexed Shield.

-- BOM --
1 * Raspberry PI model 2, 3 or 4 -> model 3 prefered, model 4 needs some extra-pin adjustments
1 * LCD 1602 -> no need for I2C interface
1 * PCM5102A -> check the jumpers, https://github.com/dwhinham/mt32-pi/wiki/GY-PCM5102-DAC-module
2 * 6.35mm stereo jack connector (J2, J5) -> ! short pin type
3 * 3.5mm stereo jack connector (J3, J4, J12) -> 1 for audio output, 2 for TRS midi, PJ307 low-profile type
1 * Diode 1N4148 (D1)
1 * Resistor 1/4w 10 ohms (R4)
1 * Resistor 1/4w 33 ohms (R3)
1 * Resistor 1/4w 220 ohms (R1) 
1 * Resistor 1/4w 470 ohms (R2)
8 * Resistor 1/4w 10 kohms (R8, R9, R10, R11, R12, R13, R14, R15)
5 * Push button (SW2, SW3, SW4, SW5, SW6 / GPIO expander board)
3 * Capacitor 10 nF (C1, C2, C3)
1 * Optocoupler H11L1 (U1) + optional 6-pin socket  
1 * Trimpot 10 kohms (RV3) -> LCD contrast adjusment
1 * Dual alpha-style potentiometer 100 kohms (RV1) -> volume control
1 * Rotary encoder (SW1 if main board, SW7 if control expander board) -> long shaft version to adjust height with potentiometer
1 * Extra-long female socket 2x20 -> connection with the Raspberry Pi
2 * Midi TRS cable -> standard midi IN / midi OUT, https://minimidi.world/
main spacers 4x 16mm between the Shield and the Pi, 2x 20mm for the left side to prevent it from tipping 
and add superlow ph3.5 sockets and pins for the 16pin LCD and the 7pin button board... need to adjust with the standoff heights, add as many spacers as you need


-- adapt minidexed.ini file --
1/ choose the right sound output
`SoundDevice=i2s`
2/ configure the buttons (as you wish)
`ButtonPinPrev=6`
`ButtonActionPrev=click`
`ButtonPinPgmUp=12`
`ButtonActionPgmUp=click`
`ButtonPinPgmDown=5`
`ButtonActionPgmDown=click`
`ButtonPinNext=7`
`ButtonActionNext=click`
`ButtonPinBack=11`
`ButtonActionBack=longpress`
`ButtonPinSelect=11`
`ButtonActionSelect=click`
`ButtonPinHome=8`
`ButtonActionHome=click`
`ButtonPinShortcut=11`

![MiniDexed_Shield-build_1](https://github.com/user-attachments/assets/bcde7c49-0e5a-4615-9b3b-dc743306167b)

![MiniDexed_Shield-build_2](https://github.com/user-attachments/assets/b5a70d28-b3ed-4816-b45e-6a6b2cc913c7)

![MiniDexed_Shield-build_3](https://github.com/user-attachments/assets/3cf8b4d3-422b-4b2f-adb8-d43eb4c0498d)

Use the extra tall 20x2 socket for the Raspberry Pi connection.

![MiniDexed_Shield-build_3_5](https://github.com/user-attachments/assets/de29c674-ca34-4a00-afeb-78f2262cd035)

The DAC does not use a socket, but is soldered directly to the board with pin headers. Take care of the height when inserting the Pi.
You may cover soldering points close to the Pi sockets with electric tape to prevent short-circuits (near "trim pins flush" label).

![MiniDexed_Shield-build_4](https://github.com/user-attachments/assets/af0ecc4e-478d-42a4-8dc4-442fc4d4b27a)

There are 3 jumper pads for configuring how the audio out is routed. Those pads may be bridged with a 0 ohm resistor, or just a blob of solder.

-- Comment from the designer --
Audio config pads, by default the stereo outs are routed to the volume control, then the headphone jack, then to the 1/4" jacks.   Inserting a 3.5 headphone or cable will disconnect the big jacks.   There is a pair of pads to change this behavior so a mini jack does not mute the big jack. Also another pad make mono summed (to L) when no R 1/4" is inserted (Dont do both of these modifications or volume will always be summed in the 3.5 stereo jack).

In my config, I let all the pads in their default position.

![MiniDexed_Shield-build_5](https://github.com/user-attachments/assets/04b9bec9-c098-43d6-80bc-b31baec588f8)

For the volume/encoder board , it needs to sit lower, so use normal long pins without the black spacer. To assemble it, first solder in the female sockets, then put the volume board directly ontop the sockets and then put individual pins through the holes into the socket. Solder from the top side then cut the pins down on the topside.

![MiniDexed_Shield-build_5_5](https://github.com/user-attachments/assets/61a73d86-733b-4e84-a0bf-6b6bf27be026)

![MiniDexed_Shield-build_6](https://github.com/user-attachments/assets/7863130b-341a-49e3-854f-cf422962642f)

![MiniDexed_Shield-build_7](https://github.com/user-attachments/assets/b681ea4d-3e88-4ac8-98b3-480ae641eb9c)


![MiniDexed_Shield_DX72_1](https://github.com/user-attachments/assets/e61f69f7-d427-4516-8116-38ad6a7faf4a)

![MiniDexed_Shield_DX72_2](https://github.com/user-attachments/assets/333ce0b4-529e-4aa2-bf4c-c1feb32b4998)

**Reply by miotislucifugis** [reply]
Hey! super work!   Thanks for the documentation! The only thing I would like to add is a note about covering the LCD pins ( the ones with a cut flush marking)  with a piece of kaptan or vinyl electrical tape so there is no possibility of the pins (especially 5v and gnd!) shorting on the pi4's usb jack.
